lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,327 @@
+        """
+        new_cube = cube.copy()
+        new_cube.rename("lightning_probability")
+        new_cube.remove_coord('threshold')
+        new_cube.attributes = {}
+        if self.debug:
+            print(('In {}, new_cube is {}'.format(self, new_cube)))
+        return new_cube
+
+    def _modify_first_guess(self, cube, fg_cube, ltng_cube, precip_cube,
+                            vii_cube):
+        """
+        Modify first-guess lightning probability with nowcast data
+
+        Args:
+            cube (iris.cube.Cube):
+                Provides the meta-data for the Nowcast lightning probability
+                output cube
+
+            fg_cube (iris.cube.Cube):
+                First-guess lightning probability
+                Must have same x & y dimensions as cube
+                Time dimension should overlap that of cube
+
+            ltng_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
+
+            precip_cube (iris.cube.Cube):
+                Nowcast precipitation probability (threshold > 0.5, 7, 35)
+                Must have same other dimensions as cube
+
+            vii_cube (iris.cube.Cube):
+                Radar-derived vertically integrated ice content (VII)
+                Must have same x and y dimensions as cube
+                Time should be a scalar coordinate
+                Must have a threshold coordinate with points matching
+                self.vii_thresholds
+                Can be <No cube> or None or anything that evaluates to False
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube containing Nowcast lightning probability.
+        """
+        new_cube_list = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over('time'):
+            thispoint = cube_slice.coord('time').points[0]
+            thistime = iris_time_to_datetime(
+                cube_slice.coord('time').copy())[0]
+            this_ltng = ltng_cube.extract(iris.Constraint(time=thistime))
+            fg_time = iris_time_to_datetime(
+                fg_cube.coord('time').copy())[
+                fg_cube.coord('time').nearest_neighbour_index(thispoint)]
+            this_fg = fg_cube.extract(iris.Constraint(time=fg_time))
+            err_string = "No matching {} cube for {}"
+            assert isinstance(this_ltng,
+                              iris.cube.Cube), err_string.format("lightning",
+                                                                 thistime)
+            assert isinstance(cube_slice,
+                              iris.cube.Cube), err_string.format("output",
+                                                                 thistime)
+            assert isinstance(this_fg,
+                              iris.cube.Cube), err_string.format("first-guess",
+                                                                 thistime)
```

Comment:
I think we would usually raise a `ValueError` here, rather than using assert statements.